flag.go

```diff
@@ -271,10 +271,13 @@ func stringifyFlag(f Flag) string {
 	needsPlaceholder := false
 	defaultValueString := ""
 	val := fv.FieldByName("Value")
+	showDefaultValue := fv.FieldByName("ShowDefaultValue")
```

Comment:
Using reflection here opens us up to potential bugs after future refactors, since we don't get any compile time enforcement that this field actually exists.

Since we're already assuming below that this field is only relevant for a bool type, what we can do instead is do a type check:
```go
showDefaultValue := true
if boolFlag, ok := f.(BoolFlag); ok {
  showDefaultValue = boolFlag.ShowDefaultValue
}
```

We then have a value that represents whether we should show the default value that works consistently across all types and is slightly more refactor proof.